src/Automation/Automation/Common/AutomationPSClientSoftwareUpdateConfiguration.cs

```diff
@@ -39,9 +39,12 @@ public SoftwareUpdateConfiguration CreateSoftwareUpdateConfiguration(string reso
                     {
 
                         var tags = new Dictionary<string, IList<string>>();
-                        foreach (var tag in query.TagSettings.Tags)
+                        if (query.TagSettings != null && query.TagSettings.Tags != null && query.TagSettings.Tags.Count > 0)
```

Comment:
@jemex nit: the last check (`query.TagSettings.Tags.Count > 0`) shouldn't be necessary here